includes/class-graduate-degree-contact-taxonomy.php

```diff
@@ -174,95 +188,18 @@ public static function get_all_term_meta( $term_id ) {
-			<th scope="row">
-				<label for="contact_title">Title</label>
-			</th>
-			<td>
-				<input type="text" name="contact_title" id="contact_title" value="<?php echo esc_attr( $term_meta['title'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_department">Department</label>
-			</th>
-			<td>
-				<input type="text" name="contact_department" id="contact_department" value="<?php echo esc_attr( $term_meta['department'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_email">Email</label>
-			</th>
-			<td>
-				<input type="text" name="contact_email" id="contact_email" value="<?php echo esc_attr( $term_meta['email'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_address_one">Address 1</label>
-			</th>
-			<td>
-				<input type="text" name="contact_address_one" id="contact_address_one" value="<?php echo esc_attr( $term_meta['address_one'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_address_two">Address 2</label>
-			</th>
-			<td>
-				<input type="text" name="contact_address_two" id="contact_address_two" value="<?php echo esc_attr( $term_meta['address_two'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_city">City</label>
-			</th>
-			<td>
-				<input type="text" name="contact_city" id="contact_city" value="<?php echo esc_attr( $term_meta['city'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_state">State</label>
-			</th>
-			<td>
-				<input type="text" name="contact_state" id="contact_state" value="<?php echo esc_attr( $term_meta['state'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_postal">Zip Code</label>
-			</th>
-			<td>
-				<input type="text" name="contact_postal" id="contact_postal" value="<?php echo esc_attr( $term_meta['postal'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_phone">Phone</label>
-			</th>
-			<td>
-				<input type="text" name="contact_phone" id="contact_phone" value="<?php echo esc_attr( $term_meta['phone'] ); ?>" />
-			</td>
-		</tr>
-		<tr class="form-field">
-			<th scope="row">
-				<label for="contact_fax">Fax</label>
-			</th>
-			<td>
-				<input type="text" name="contact_fax" id="contact_fax" value="<?php echo esc_attr( $term_meta['fax'] ); ?>" />
-			</td>
-		</tr>
-		<?php
+
+		foreach( $this->term_meta_keys as $key => $meta ) {
+			?>
+			<tr class="form-field">
+				<th scope="row">
+					<label for="<?php echo esc_attr( $key ); ?>"><?php echo esc_html( $meta['description'] ); ?></label>
+				</th>
+				<td>
+					<input type="text" name="<?php echo esc_attr( $key ); ?>" id="<?php echo esc_attr( $key ); ?>" value="<?php echo esc_attr( $term_meta[ $key ][0] ); ?>" />
```

Comment:
The `value` output here pops an `undefined index` notice when editing a newly-added contact.